pulp_ansible/app/tasks/signature.py

```diff
@@ -22,6 +23,38 @@
 from pulpcore.plugin.models import SigningService, ProgressReport
 from pulpcore.plugin.sync import sync_to_async_iterable, sync_to_async
 from pulp_ansible.app.tasks.utils import get_file_obj_from_tarball
+from rest_framework import serializers
+
+
+def verify_signature_upload(data):
+    """The task code for verifying collection signature upload."""
+    file = data["file"]
+    sig_data = file.read()
+    file.seek(0)
+    collection = data["signed_collection"]
+    keyring = None
+    if (repository := data.get("repository")) and repository.keyring:
+        keyring = repository.keyring
+    gpg = gnupg.GPG(keyring=keyring)
```

Comment:
looks like this line should have been 
```py
gpg = gnupg.GPG(gnupghome=settings.ANSIBLE_CERTS_DIR, keyring=keyring)
```

@gerrod3

Reply:
Can we please try to not make the gpg operations plugin specific? This will end in a configuration nightmare.
Maybe we should even provide a `pulpcore.plugin` interface that wraps around `gnupg.GPG` to make it consistent for plugins.
Plugins to bring to the table include ansible, container, deb and rpm at this time.